src/esw/science.py

```diff
@@ -171,105 +168,54 @@ def handle_change_servo_angles(self, req: ChangeServoAngleRequest) -> ChangeServ
         :returns: A boolean that is the success of sent UART transaction.
         """
         success = self._send_mosfet_msg("uv_led_carousel", req.enable)
         return ChangeDeviceStateResponse(success)
 
     def handle_change_uv_led_end_effector_state(self, req: ChangeDeviceStateRequest) -> ChangeDeviceStateResponse:
-        """Processes a request to change the UV LED on the end effector by
-        issuing the command to the STM32 chip via UART. Returns the success of
-        transaction.
+        """Process a request to change the UV LED on the end effector by
+        issuing the command to the STM32 chip via UART.
         :param req: A boolean that is the requested UV LED state.
         :returns: A boolean that is the success of sent UART transaction.
         """
         success = self._send_mosfet_msg("uv_led_end_effector", req.enable)
         return ChangeDeviceStateResponse(success)
 
     def handle_change_white_led_state(self, req: ChangeDeviceStateRequest) -> ChangeDeviceStateResponse:
-        """Processes a request to change the carousel white LED by
-        issuing the command to the STM32 chip via UART. Returns the success of
-        transaction.
+        """Process a request to change the carousel white LED by
+        issuing the command to the STM32 chip via UART.
         :param: req: A boolean that is the requested white LED state.
         :returns: A boolean that is the success of sent UART transaction.
         """
         success = self._send_mosfet_msg("white_led", req.enable)
         return ChangeDeviceStateResponse(success)
 
-    def receive(self) -> None:
-        """Reads in a message from the UART RX line and processes it. If there
-        is data, then the data will be published to the corresponding ROS
-        topic.
-        Depending on the NMEA tag of the message, the proper function
-        will be called to process the message.
-        """
-        rx_msg = self._read_msg()
-        arr = rx_msg.split(",")
-
-        tag = arr[0][3:]
-        if not (tag in self._handler_function_by_tag):
-            rospy.sleep(self._sleep_amt)
-            return
-
-        self._handler_function_by_tag[tag](rx_msg)
-
     def _auton_led_transmit(self, color: str) -> bool:
-        """Sends a UART message to the STM32 commanding the auton LED array
+        """Send a UART message to the STM32 commanding the auton LED array
         state.
         :param color: A string that is the color of the requested state of
             the auton LED array. Note that green actually means blinking green.
             The string should be lower case.
         :returns: A boolean that is the success of the transaction. Note that
             this could be because an invalid color was sent.
         """
-        assert color.islower(), "color should be lower case"
+        # TODO: Add LED functionality similar to master.
+        color = color.lower()
```

Comment:
wdym by todo

Reply:
if ur talking about the blinking green lights, then that gets handled on the nucleo end.

Reply:
Okay I'm sure you're right cause idk what the final architecture is gonna be, but can you explain it to me? I was thinking we need to give this the same functionality as the temp solution that's in master right now. Also I thought an arduino was controlling the led and that it didn't have the blinking functionality.

Reply:
ok i'll accept this pr but just make sure to remove comment